### PR TITLE
test: testMachines: Don't stress libvirtd on teardowns

### PR DESCRIPTION
**Pull request by KKoukiou**

### CODE REVIEW

File: test/verify/check-machines [hunk 1]
```diff
@@ -123,6 +123,10 @@ class TestMachines(MachineCase, StorageHelpers, NetworkHelpers):
         self.addCleanup(self.browser.logout)
         self.allow_authorize_journal_messages()
 
+        wait(lambda: "4" in self.machine.execute("virsh net-list --all | wc -l"), delay=3)
+        wait(lambda: "3" in self.machine.execute("virsh pool-list --all | wc -l"), delay=3)
+        wait(lambda: "3" in self.machine.execute("virsh list --all | wc -l"), delay=3)
```
Comment (martinpitt) [inline test/verify/check-machines:128]: These magic numbers are awkward, let's please not make assumptions about the table formatting. Please call these with `--name`, append `.strip()`, and compare for exact equality `== '0`'. We don't want this to succeed if there are still 13 domains or so.

File: test/verify/check-machines [hunk 2]
```diff
@@ -2770,6 +2770,9 @@ class TestMachines(MachineCase, StorageHelpers, NetworkHelpers):
         b = self.browser
         m = self.machine
 
+        m.execute("ip link add virbr2 type bridge")
```
Comment (martinpitt) [inline test/verify/check-machines:2793]: How sure are you that virbr2 won't collide with anything that libvirt could create by itself? Could it be something more specific, like "virbrtest" or so?

File: test/verify/check-machines [hunk 3]
```diff
@@ -778,7 +778,6 @@ class TestMachines(MachineCase, StorageHelpers, NetworkHelpers):
 
             m.execute("virsh pool-define-as iscsi-pool --type iscsi --target /dev/disk/by-id --source-host 127.0.0.1 --source-dev {0} && virsh pool-start iscsi-pool".format(target_iqn))
             wait(lambda: "unit:0:0:0" in self.machine.execute("virsh pool-refresh iscsi-pool && virsh vol-list iscsi-pool"), delay=3)
-
```
Comment (martinpitt) [inline test/verify/check-machines:776]: In this "tests: testMachines:tesStorageVolumesCreate remove unecessary pool cleanup" commit, can you please fix the tes*t*Storage typo? Most typos/grammar issues in the commit logs are not important, but one often greps for test names/identifiers in `git log`.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Very nice work, thanks! I suspected this to be more brittle than a stop+rm -rf, but if that causes too much trouble, let's try this approach. Kudos for the lots of extra fixes/robustifications!

I found a few small issues.

There are still [machines failures](https://logs.cockpit-project.org/logs/pull-14919-20201124-084829-84a2a57f-fedora-33/log.html#309), they may be unrelated, of course.

File: test/verify/check-machines (hunk 2 above)
Comment (KKoukiou) [reply test/verify/check-machines:2793]: I can do that - I thought virbr0 only is libvirt created but good point.

File: test/verify/check-machines (hunk 1 above)
Comment (KKoukiou) [reply test/verify/check-machines:128]: --name of course, good point

**Comment by KKoukiou** [comment]
@martinpitt all the above are known flakes - you can see also in the failure stats. I think I might have a solution to the super common testDiskEdit flake so I will repost with one more try if you prefer to see it more green.

**Comment by martinpitt** [comment]
A little too many machines failures, another round of testing.

**Comment by martinpitt** [comment]
The [testAddDiskDirPool failure on ubuntu-2004](https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/pull-14919-20201124-155851-a1cc5fb3-ubuntu-2004/log.html#268) also [happens on #14955](https://logs.cockpit-project.org/logs/pull-14955-20201124-155623-5bd956c7-ubuntu-2004/log.html), so is unrelated. Thanks!

**Review by martinpitt — DISMISSED** [review]

**Comment by KKoukiou** [comment]
Hmm @martinpitt I will not merge it yet, debian-stable has some weird failures totally related to the changed. Maybe I will try to split this into smaller PRs as well.

**Comment by KKoukiou** [comment]
This was left a bit behind by me, sorry :( I rebased and will check the errors on CI. last run seemed good actually but it was already conflicting.

**Comment by KKoukiou** [comment]
I have put quite some effort here, but it did not help the coredumps really. 
I will close it because I have reached a dead end with no ideas on this, and will try to readdress it in the future in another way.

**Pull request closed without merging**